src/core/type-representation.ts

```diff
@@ -99,7 +99,11 @@ const getObjectTypeRepresentation = (entryType: tsMorph.Type, source: tsMorph.So
 
         const propertyType = property.getTypeAtLocation(source);
         const propertyTypeRepresentation = getTypeRepresentation(propertyType, source);
-        if (typeMatcher && exampleRepresentation) {
+
+        // recursively set type representation for nested object properties
+        if (propertyType.isObject() && !propertyType.isArray()) {
```

Comment:
It is not solution to this problem.

The problem is in line #90 https://github.com/HLTech/pact-gen-ts/pull/60/files#diff-c6a2f1cd0cfa639da00a7d6c3dfb23df5d133f76e94823ec2619cd05535aa917R90 of this file. We get all descendants of object property (so, if AST contains some JSDocTag node for nested field of some parent field, this is persisted only for parent object field). So, the solution is getting only children nodes of JSDocComments kind of current field.

Working solution:
```ts
        const jsDocTagsOfProperty = property.getValueDeclaration()?.getChildrenOfKind(ts.SyntaxKind.JSDocComment).reduce((jsDocTags, jsDocCommentElement) => {
            return [...jsDocTags, ...(jsDocCommentElement.getChildrenOfKind(ts.SyntaxKind.JSDocTag))];
        }, []);
```
use it, and remove your code :)